lib/active_model/serializer/utils.rb

```diff
@@ -0,0 +1,42 @@
+module ActiveModel::Serializer::Utils
+  module_function
+
+    # converts the include hash to a standard format
+    # for constrained serialization recursion
+    #
+    # converts
+    #  [:author, :comments => [:author]] to
+    #  {:author => [], :comments => [:author]}
+    #
+    # and
+    #  [:author, :comments => {:author => :bio}, :posts => [:comments]] to
+    #  {:author => [], :comments => {:author => :bio}, :posts => [:comments]}
+    #
+    # The data passed in to this method should be an array where the last
+    # parameter is a hash
+    #
+    # the point of this method is to normalize the include
+    # options for the child relationships.
+    # if a sub inclusion is still an array after this method,
+    # it will get converted during the next iteration
+    def include_array_to_hash(include_array)
```

Comment:
It's tricky cause the nested config could either be a symbol or array. This just takes all the possibilities and makes them consistent for internal use